app.js

```diff
@@ -1,32 +1,55 @@
-const validations = require('./api/controllers/v0.1/validations.js');
-const createPatronV0_2 = require('./api/controllers/v0.2/createPatron.js'); // eslint-disable-line camelcase
-const apiDoc = require('./api/controllers/apiDoc.js');
-const createPatronV0_3 = require('./api/controllers/v0.3/endpoints.js'); // eslint-disable-line camelcase
+const createPatronV0_1 = require("./api/controllers/v0.1/createPatron.js"); // eslint-disable-line camelcase
+const validations = require("./api/controllers/v0.1/validations.js");
+const createPatronV0_2 = require("./api/controllers/v0.2/createPatron.js"); // eslint-disable-line camelcase
+const apiDoc = require("./api/controllers/apiDoc.js");
+const createPatronV0_3 = require("./api/controllers/v0.3/endpoints.js"); // eslint-disable-line camelcase
+
+const awsDecrypt = require("./config/awsDecrypt");
 
 const app = express();
-const pathName = `${process.cwd()}/config/deploy_${app.get('env')}.env`;
-require('dotenv').config({ path: pathName });
-
-const BarcodeDb = require('./db');
-
-// Initialize the connection to the database.
-// This is done here so we can have one instance and one connection to the
-// database for however many times the lambda is used for.
-const db = BarcodeDb({
-  user: process.env.DB_USER,
-  host: process.env.DB_HOST,
-  database: process.env.DB_DATABASE,
-  password: process.env.DB_PASSWORD,
-  port: process.env.DB_PORT,
-});
-// Once we get the database class instance, initialize it by creating the
-// table and inserting the seed data if it's not in the database already.
-db.init();
+const pathName = `${process.cwd()}/config/deploy_${app.get("env")}.env`;
+require("dotenv").config({ path: pathName });
+
+const BarcodeDb = require("./db");
+
+// Decrypt any database credentials if on QA or production for the NYPL AWS
+// environment. Then create the single database instance in the app.
+async function initDatabase() {
+  let dbUser;
+  let dbHost;
+  let dbPassword;
+
+  if (process.env.NODE_ENV === "development") {
+    dbUser = process.env.DB_USER;
+    dbHost = process.env.DB_HOST;
+    dbPassword = process.env.DB_PASSWORD;
+  } else {
+    dbUser = await awsDecrypt.decryptKMS(process.env.DB_USER);
+    dbHost = await awsDecrypt.decryptKMS(process.env.DB_HOST);
+    dbPassword = await awsDecrypt.decryptKMS(process.env.DB_PASSWORD);
+  }
+
+  // Initialize the connection to the database.
+  // This is done here so we can have one instance and one connection to the
+  // database for however many times the lambda is used for.
+  const db = BarcodeDb({
+    user: dbUser,
+    host: dbHost,
+    password: dbPassword,
+    // These are constant and don't need to be encrypted or decrypted.
+    database: process.env.DB_DATABASE,
+    port: process.env.DB_PORT,
+  });
+  // Once we get the database class instance, initialize it by creating the
+  // table and inserting the seed data if it's not in the database already.
+  db.init();
+}
+
+initDatabase();
```

Comment:
I had to make this a function because it's easier to get the asynchronous decrypted values through an async/await in a function.